packages/client/src/upload.ts

```diff
@@ -87,6 +96,27 @@ export async function* upload(
 
         const { data } = file;
 
+        uploadProgress.bytesUploaded = 0;
+
+        // Split out into chunks
+        const body = new Readable();
+        const originalRead = body.read.bind(body);
+        body.read = function (...args) {
+          const chunk = originalRead(...args);
+          if (chunk) {
+            uploadProgress.bytesUploaded += chunk.length;
+            uploadProgress.emit('progress');
+          }
+          return chunk;
+        };
+
+        const chunkSize = 16384; /* 16kb - default Node.js `highWaterMark` */
+        for (let i = 0; i < data.length; i += chunkSize) {
+          const chunk = data.slice(i, i + chunkSize);
```

Comment:
This can probably be optimized because each call to `.slice()` is going to iterate over the data again. Its probably negligible for small `data` but as it gets larger it might be noticeable.

Reply:
I'm not sure what you mean that `.slice()` is going to iterate over the data. It basically just creates a new Buffer "view" ([source](https://github.com/nodejs/node/blob/main/lib/buffer.js#L1122-L1132)). Should be very fast. Are you thinking of `Array#slice()` behavior?

Reply:
Yes, I am indeed thinking of `Array#slice()` 😅 

Thanks for clarifying 👍